hack/test-e2e-provider-local.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+repo_root="$(readlink -f $(dirname ${0})/..)"
+
+if [[ ! -d "$repo_root/gardener" ]]; then
+  git clone https://github.com/gardener/gardener.git
+fi
+
+cd "$repo_root/gardener"
+git checkout 20b05ad20e6f426e65e7bd904879a77a94576b02 # g/g v1.66.1
+make kind-up
+export KUBECONFIG=$repo_root/gardener/example/gardener-local/kind/local/kubeconfig
+make gardener-up
+
+cd $repo_root
+
+version=$(git rev-parse HEAD)
+make docker-images
+docker tag eu.gcr.io/gardener-project/gardener/extensions/networking-cilium:latest networking-cilium-local:$version
+kind load docker-image networking-cilium-local:$version --name gardener-local
+
+mkdir -p $repo_root/tmp
+cp -f $repo_root/example/controller-registration.yaml $repo_root/tmp/controller-registration.yaml
+yq -i e "(select (.providerConfig.values.image) | .providerConfig.values.image.tag) |= \"$version\"" $repo_root/tmp/controller-registration.yaml
+yq -i e '(select (.providerConfig.values.image) | .providerConfig.values.image.repository) |= "docker.io/library/networking-cilium-local"' $repo_root/tmp/controller-registration.yaml
+
+kubectl apply -f "$repo_root/tmp/controller-registration.yaml"
+
+# reduce flakiness in contended pipelines
+export GOMEGA_DEFAULT_EVENTUALLY_TIMEOUT=5s
+export GOMEGA_DEFAULT_EVENTUALLY_POLLING_INTERVAL=200ms
+# if we're running low on resources, it might take longer for tested code to do something "wrong"
+# poll for 5s to make sure, we're not missing any wrong action
+export GOMEGA_DEFAULT_CONSISTENTLY_DURATION=5s
+export GOMEGA_DEFAULT_CONSISTENTLY_POLLING_INTERVAL=200ms
+
+GO111MODULE=on ginkgo --timeout=1h --v --progress "$@" $repo_root/test/e2e/...
```

Comment:
Should we not clean up after the test, e.g. `make gardener-down && make kind-down`?